### PR TITLE
fix(node): various improvements to the node binary

### PR DESCRIPTION
**Pull request by mlegner**
- remove unnecessary serde attributes
- minor improvements to the `key-gen` command
- move options for `generate-config` to new `GenerateConfigArgs` struct
- make votes for storage and write price optional

### CODE REVIEW

File: crates/walrus-service/bin/deploy.rs [hunk 1]
```diff
@@ -75,11 +78,11 @@ struct DeploySystemContractArgs {
     testbed_config_path: Option<PathBuf>,
     // Note: The storage unit is set in `crates/walrus-sui/utils.rs`. Change the unit in
     // the doc comment here if it changes.
-    /// The price to set per unit of storage (1 KiB) and epoch.
-    #[arg(long, default_value_t = 50)]
+    /// The price in FROST to set per unit of storage (1 KiB) and epoch.
```
Comment (halfprice) [inline crates/walrus-service/bin/deploy.rs]:
```suggestion
    /// The price in FROST to set per unit of storage (1 KiB) per epoch.
```

File: crates/walrus-service/bin/deploy.rs [hunk 2]
```diff
@@ -75,11 +78,11 @@ struct DeploySystemContractArgs {
...
+    #[arg(long, default_value_t = config::defaults::storage_price())]
     storage_price: u64,
-    /// The price to set for writing one unit of storage (1 KiB).
-    #[arg(long, default_value_t = 50)]
+    /// The price in FROST to set for writing one unit of storage (1 KiB).
```
Comment (halfprice) [inline crates/walrus-service/bin/deploy.rs:84]:
```suggestion
    /// The price in FROST to set for writing one unit of storage (1 KiB) per epoch.
```

File: crates/walrus-service/bin/node.rs [hunk 3]
```diff
@@ -62,92 +62,115 @@ enum Commands {
...
+        force: bool,
     },
 
+    /// Generate a new node configuration.
+    GenerateConfig(Box<GenerateConfigArgs>),
+
     /// Register a new node with the Walrus storage network.
     Register {
         #[clap(short, long)]
-        #[serde(deserialize_with = "crate::utils::resolve_home_dir")]
```
Comment (halfprice) [inline crates/walrus-service/bin/node.rs:79]: Just give this a try and it is indeed not needed. Does this mean that we can remove `resolve_home_dir` anywhere?

File: crates/walrus-service/bin/node.rs [hunk 4]
```diff
@@ -62,92 +62,115 @@ enum Commands {
...
-        #[clap(default_value = "protocol.key")]
-        out: PathBuf,
-        /// Whether to generate a network key instead of a protocol key.
-        #[clap(short, long, default_value_t = false)]
-        network_key: bool,
+        /// not overwritten and the operation will fail unless the `--force` option is provided.
+        /// [default: ./<KEY_TYPE>.key]
+        #[clap(short, long)]
+        out: Option<PathBuf>,
+        /// Which type of key to generate.
```
Comment (halfprice) [inline crates/walrus-service/bin/node.rs]: suggestion: can we print the two options?

File: crates/walrus-service/bin/node.rs [hunk 5]
```diff
@@ -322,36 +361,32 @@ mod commands {
...
+            voting_params: VotingParams {
+                storage_price,
+                write_price,
+                node_capacity,
+            },
+            rest_api_address,
+            metrics_address,
+            name,
+            commission_rate,
+            ..Default::default()
```
Comment (halfprice) [inline crates/walrus-service/bin/node.rs:390]: Nice! Should have thought about this...

File: crates/walrus-service/src/node/config.rs [hunk 6]
```diff
@@ -367,6 +367,16 @@ pub mod defaults {
         Some(Some(REST_GRACEFUL_SHUTDOWN_PERIOD_SECS))
     }
 
+    /// The default vote for the storage price.
+    pub fn storage_price() -> u64 {
+        5
```
Comment (halfprice) [inline crates/walrus-service/src/node/config.rs:372]: Question: I thought about doing this but don't know how to set the default properly. 5 FROST seems too cheap...

File: crates/walrus-service/bin/deploy.rs (hunk 2 above)
Comment (mlegner) [reply crates/walrus-service/bin/deploy.rs:84]: This is a one-time price, so there should not be an epoch.

File: crates/walrus-service/bin/node.rs (hunk 3 above)
Comment (mlegner) [reply crates/walrus-service/bin/node.rs:79]: The serde attributes are only necessary in the client binary when using the JSON mode. As we don't have that in the node binary, we don't need any serde attributes.

File: crates/walrus-service/src/node/config.rs (hunk 6 above)
Comment (mlegner) [reply crates/walrus-service/src/node/config.rs:372]: Assuming 1 WAL ~ 1 USD, this value of 5 FROST / KiB / epoch corresponds to ~ 5 * 12 * 1M FROST ~ 0.06 USD per GiB (encoded!) per year, which corresponds to ~ 0.3 USD per GiB (unencoded) per year. So this seems quite reasonable to me.

File: crates/walrus-service/bin/node.rs (hunk 3 above)
Comment (halfprice) [reply crates/walrus-service/bin/node.rs:79]: Good to know!

File: crates/walrus-service/src/node/config.rs (hunk 6 above)
Comment (halfprice) [reply crates/walrus-service/src/node/config.rs:372]: The alpha here is `Assuming 1 WAL ~ 1 USD` :)

**Pull request merged**